include/deal.II/fe/fe_interface_values.h

```diff
@@ -1037,6 +1418,324 @@ FEInterfaceValues<dim, spacedim>::jump_3rd_derivative(
+    for (int d = 0; d < dim; ++d)
+      result[d] =
+        this->fe_interface->average(interface_dof_index,
+                                    q_point,
+                                    d + extractor.first_vector_component);
+
+    return result;
+  }
+
+
+
+  template <int dim, int spacedim>
+  typename Vector<dim, spacedim>::gradient_type
+  Vector<dim, spacedim>::average_gradient(
+    const unsigned int interface_dof_index,
+    const unsigned int q_point) const
+  {
+    gradient_type result;
+    for (int d = 0; d < dim; ++d)
+      result[d] = this->fe_interface->average_gradient(
+        interface_dof_index, q_point, d + extractor.first_vector_component);
+
+    return result;
+  }
+
+
+
+  template <int dim, int spacedim>
+  typename Vector<dim, spacedim>::gradient_type
+  Vector<dim, spacedim>::jump_gradient(const unsigned int interface_dof_index,
+                                       const unsigned int q_point) const
+  {
+    gradient_type result;
+    for (int d = 0; d < dim; ++d)
+      result[d] =
+        this->fe_interface->jump_gradient(interface_dof_index,
+                                          q_point,
+                                          d + extractor.first_vector_component);
+
+    return result;
+  }
+
+
+
+  template <int dim, int spacedim>
+  typename Vector<dim, spacedim>::hessian_type
+  Vector<dim, spacedim>::average_hessian(const unsigned int interface_dof_index,
+                                         const unsigned int q_point) const
+  {
+    hessian_type result;
+    for (int d = 0; d < dim; ++d)
+      result[d] = this->fe_interface->average_hessian(
+        interface_dof_index, q_point, d + extractor.first_vector_component);
+
+    return result;
+  }
+
+
+
+  template <int dim, int spacedim>
+  typename Vector<dim, spacedim>::hessian_type
+  Vector<dim, spacedim>::jump_hessian(const unsigned int interface_dof_index,
+                                      const unsigned int q_point) const
+  {
+    hessian_type result;
+    for (int d = 0; d < dim; ++d)
+      result[d] =
+        this->fe_interface->jump_hessian(interface_dof_index,
+                                         q_point,
+                                         d + extractor.first_vector_component);
+
+    return result;
+  }
+
+
+
+  template <int dim, int spacedim>
+  typename Vector<dim, spacedim>::third_derivative_type
+  Vector<dim, spacedim>::jump_3rd_derivative(
+    const unsigned int interface_dof_index,
+    const unsigned int q_point) const
+  {
+    third_derivative_type result;
+    for (int d = 0; d < dim; ++d)
+      result[d] = this->fe_interface->jump_3rd_derivative(
+        interface_dof_index, q_point, d + extractor.first_vector_component);
+
+    return result;
+  }
+} // namespace FEInterfaceViews
+
+
+
+namespace internal
+{
+  namespace FEInterfaceViews
+  {
+    template <int dim, int spacedim>
+    Cache<dim, spacedim>::Cache(
+      const FEInterfaceValues<dim, spacedim> &fe_interface)
```

Comment:
This function should probably go into the `.cc` file if we want to keep the cache at all.

Reply:
@zjiaqi2018 you will need to do the following:
1. create source/fe/fe_interface_values.cc (it doesn't exist yet, you can steal the structure/header from one of the other files)
2. Move the implementation of Cache::Cache() (where this comment is) into the .cc file
3. add a fe_interface_value.is.inst.in (include in the .cc) and instantiate the Cache for all pairs of dim and spacedim, see fe_dgp.inst.in for an example.
Let me know if you need more help with that.

Reply:
Actually, scratch that and ignore what I wrote. Too much effort as both of them asked to remove the Cache class anyways.

Can you try to:
1. change ``operator[]`` to return the views by value instead of by reference.
2. In the implementation of operator[], create the ``FEInterfaceViews::Scalar<dim, spacedim>`` and return it immediately.
3. Get rid of everything about ``Cache``

Reply:
@tjhei  Updated, please let me know if something needs to be changed.